### PR TITLE
fix: restrict 'user ban' functionality to member with ChapterBanUser rule

### PR DESCRIPTION
**Pull request by shootermv**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Im adding here cypress test that should check - that admin of chapter1 cannot ban members of chapter2,
if he tries to do that - he should see 'access denied' notification
Closes part of #1217

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey @shootermv, thanks for working on this. This should be a `ChapterPermission`, since a chapter admin should be able to ban people from their chapter, but no others.

Also, could you create a test that confirms this, that would be awesome.  The `rejects chapter admin from changing chapter user role` spec should be a good starting point, but with `'admin@of.chapter.two'` instead.  It will also need a `banChaperUser` Cypress command, obviously!

**Comment by shootermv** [comment]
as far as i understood, test for `admin@of.chpater.two` not bale to ban user of chapter one need to be added, im i right?

File: server/prisma/generator/factories/instanceRoles.factory.ts [hunk 1]
```diff
@@ -20,6 +24,7 @@ const roles: InstanceRole[] = [
     permissions: [
       InstancePermission.ChapterJoin,
       InstancePermission.ChapterSubscriptionsManage,
+      ChapterPermission.ChapterBanUser,
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/instanceRoles.factory.ts:27]:
```suggestion
```
Members should not be able to ban anyone.

File: server/prisma/generator/factories/instanceRoles.factory.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { prisma } from '../../../src/prisma';
 
-import { InstancePermission, Permission } from '../../../../common/permissions';
+import {
+  ChapterPermission,
+  InstancePermission,
+  Permission,
+} from '../../../../common/permissions';
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/instanceRoles.factory.ts:7]:
```suggestion
import { InstancePermission, Permission } from '../../../../common/permissions';
```
See below

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
> as far as i understood, test for admin@of.chpater.two not able to ban user of chapter one need to be added, Am i right?

Yes, please!

File: cypress/e2e/dashboard/chapters/users.cy.ts [hunk 3]
```diff
@@ -140,6 +140,24 @@ describe('Chapter Users dashboard', () => {
       .as('firstUnbannedMember');
   }
 
+  it('someone whos NOT an admin of a chapter, should NOT be able to ban a chapter users', () => {
+    cy.logout();
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/users.cy.ts:144]:
```suggestion
```
`cy.login` changes the session, so there's no need to log out first.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This is looking good, just the one nitpick:

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Thanks @shootermv :rocket:

**Pull request merged**